### ISSUE TITLE
False positive with Style/IfWithBooleanLiteralBranches and nonzero?

### ISSUE THREAD

**Issue by erwanlr** [rubocop-hq/rubocop#9459]
The `nonzero?` method returns self if num is not zero, nil otherwise, and not a boolean. Causing false positive with the `Style/IfWithBooleanLiteralBranches` cop with code such as `val.nonzero? ? true : false`

--------

## Expected behavior

The cop should not flag statements such as `val.nonzero? ? true : false`

## Actual behavior

```
 Style/IfWithBooleanLiteralBranches: Remove redundant ternary operator with boolean literal branches.
      NS::Browser.get(url(path)).code.nonzero? ? true : false
```

## Steps to reproduce the problem

Run the cop against code like `val.nonzero? ? true : false` , for example https://github.com/wpscanteam/CMSScanner/blob/master/lib/cms_scanner/web_site.rb#L74

## RuboCop version

```
1.9.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 2.7.2 x86_64-linux)
  - rubocop-performance 1.9.2
```
